lsp-mode.el

```diff
@@ -5171,9 +5171,18 @@ It will show up only if current point has signature help."
 
 (lsp-defun lsp--execute-command ((action &as &Command :command :arguments?))
   "Parse and execute a code ACTION represented as a Command LSP type."
-  (-if-let* ((action-handler (lsp--find-action-handler command)))
-      (funcall action-handler action)
-    (lsp--send-execute-command command arguments?)))
+  (let ((server-id (->> (lsp-workspaces)
+                        (cl-first)
+                        (or lsp--cur-workspace)
+                        (lsp--workspace-client)
+                        (lsp--client-server-id))))
+    (condition-case nil
+        (prog1 (lsp-execute-command server-id (intern command) arguments?)
+          (lsp--warn "`lsp-execute-command' is deprecated"))
```

Comment:
Why don't you just make `lsp-execute-command` obsolete instead? We will have compile warning about that, and that would be standard way to alert package maintainer about obsolete function.

Reply:
That somehow slipped my mind; concluded from testing, calling that function doesn't cause warnings, but overriding it using `cl-defmethod` does. I couldn't find anything in the Emacs docs about the obsoletion of generics, though.